### PR TITLE
Fill object defaults for blank or missing fields and fix Site:GroundReflectance:SnowModifier defaults

### LINKED ISSUE #10282: Supplemental heating coil default not filled in
Issue overview
--------------
In the object `ZoneHVAC:TerminalUnit:VariableRefrigerantFlow`, the `Maximum Outdoor Dry-Bulb Temperature for Supplemental Heater Operation` is not filled with the default value if it's not present in the object, as the min-field=19 does not cover this field. This would mean if the line corresponding to "Maximum Outdoor Dry-Bulb Temperature for Supplemental Heater Operation" is present and the user didn't specify a value for it, the supplemental heater will turn off at 21C, but if this line is not present, then the supplemental heater will turn off at 0C.

```
  N12, \field Maximum Outdoor Dry-Bulb Temperature for Supplemental Heater Operation
       \type real
       \maximum 21.0
       \default 21.0
       \units C
       \note Supplemental heater will not operate when outdoor temperature exceeds this value.
```

The issue is examined with two files adapted from VariableRefrigerantFlow_wSuppHeater_5Zone.idf. The capacity of TU1 VRF DX Heating Coil is changed from autosize to 500 (smaller than the demand) so that supplemental coil is needed for longer period. The runperiod is changed to 1/1 to 2/4.

- In this idf file, the "Maximum Outdoor Dry-Bulb Temperature for Supplemental Heater Operation" line is not present
[VariableRefrigerantFlow_wSuppHeater_5Zone.idf.txt](https://github.com/NREL/EnergyPlus/files/13193452/VariableRefrigerantFlow_wSuppHeater_5Zone.idf.txt)

The TU1 looks like this.
```
  ZoneHVAC:TerminalUnit:VariableRefrigerantFlow,
    TU1,                     !- Zone Terminal Unit Name
    VRFAvailSched,           !- Terminal Unit Availability Schedule
    TU1 Inlet Node,          !- Terminal Unit Air Inlet Node Name
    TU1 Outlet Node,         !- Terminal Unit Air Outlet Node Name
    autosize,                !- Cooling Supply Air Flow Rate {m3/s}
    autosize,                !- No Cooling Supply Air Flow Rate {m3/s}
    autosize,                !- Heating Supply Air Flow Rate {m3/s}
    autosize,                !- No Heating Supply Air Flow Rate {m3/s}
    autosize,                !- Cooling Outdoor Air Flow Rate {m3/s}
    autosize,                !- Heating Outdoor Air Flow Rate {m3/s}
    autosize,                !- No Load Outdoor Air Flow Rate {m3/s}
    VRFFanSchedule,          !- Supply Air Fan Operating Mode Schedule Name
    drawthrough,             !- Supply Air Fan Placement
    Fan:SystemModel,         !- Supply Air Fan Object Type
    TU1 VRF Supply Fan,      !- Supply Air Fan Object Name
    OutdoorAir:Mixer,        !- Outside Air Mixer Object Type
    TU1 OA Mixer,            !- Outside Air Mixer Object Name
    COIL:Cooling:DX:VariableRefrigerantFlow,  !- Cooling Coil Object Type
    TU1 VRF DX Cooling Coil, !- Cooling Coil Object Name
    COIL:Heating:DX:VariableRefrigerantFlow,  !- Heating Coil Object Type
    TU1 VRF DX Heating Coil, !- Heating Coil Object Name
    30,                      !- Zone Terminal Unit On Parasitic Electric Energy Use {W}
    20,                      !- Zone Terminal Unit Off Parasitic Electric Energy Use {W}
    ,                        !- Rated Heating Capacity Sizing Ratio {W/W}
    ,                        !- Availability Manager List Name
    ,                        !- Design Specification ZoneHVAC Sizing Object Name
    Coil:Heating:Electric,   !- Supplemental Heating Coil Object Type
    TU1 Supp Heating Coil,   !- Supplemental Heating Coil Name
    autosize;                !- Maximum Supply Air Temperature from Supplemental Heater {C}
```
In this case, the supplemental heating coil turns off at 0C

image

- When the field is explicitly added in the idf, 
[VariableRefrigerantFlow_wSuppHeater_5Zone_explicitField.idf.txt](https://github.com/NREL/EnergyPlus/files/13193472/VariableRefrigerantFlow_wSuppHeater_5Zone_explicitField.idf.txt)
where the TU1 looks like this

```
  ZoneHVAC:TerminalUnit:VariableRefrigerantFlow,
    TU1,                     !- Zone Terminal Unit Name
    VRFAvailSched,           !- Terminal Unit Availability Schedule
    TU1 Inlet Node,          !- Terminal Unit Air Inlet Node Name
    TU1 Outlet Node,         !- Terminal Unit Air Outlet Node Name
    autosize,                !- Cooling Supply Air Flow Rate {m3/s}
    autosize,                !- No Cooling Supply Air Flow Rate {m3/s}
    autosize,                !- Heating Supply Air Flow Rate {m3/s}
    autosize,                !- No Heating Supply Air Flow Rate {m3/s}
    autosize,                !- Cooling Outdoor Air Flow Rate {m3/s}
    autosize,                !- Heating Outdoor Air Flow Rate {m3/s}
    autosize,                !- No Load Outdoor Air Flow Rate {m3/s}
    VRFFanSchedule,          !- Supply Air Fan Operating Mode Schedule Name
    drawthrough,             !- Supply Air Fan Placement
    Fan:SystemModel,         !- Supply Air Fan Object Type
    TU1 VRF Supply Fan,      !- Supply Air Fan Object Name
    OutdoorAir:Mixer,        !- Outside Air Mixer Object Type
    TU1 OA Mixer,            !- Outside Air Mixer Object Name
    COIL:Cooling:DX:VariableRefrigerantFlow,  !- Cooling Coil Object Type
    TU1 VRF DX Cooling Coil, !- Cooling Coil Object Name
    COIL:Heating:DX:VariableRefrigerantFlow,  !- Heating Coil Object Type
    TU1 VRF DX Heating Coil, !- Heating Coil Object Name
    30,                      !- Zone Terminal Unit On Parasitic Electric Energy Use {W}
    20,                      !- Zone Terminal Unit Off Parasitic Electric Energy Use {W}
    ,                        !- Rated Heating Capacity Sizing Ratio {W/W}
    ,                        !- Availability Manager List Name
    ,                        !- Design Specification ZoneHVAC Sizing Object Name
    Coil:Heating:Electric,   !- Supplemental Heating Coil Object Type
    TU1 Supp Heating Coil,   !- Supplemental Heating Coil Name
    autosize,                !- Maximum Supply Air Temperature from Supplemental Heater {C}
    ;                        !- Maximum Outdoor Dry-Bulb Temperature for Supplemental Heater Operation <--- explicitly added
```
Then the supplemental turn-off temperature is above 0C (it's 21C).

image

### Details
Some additional details for this issue (if relevant):
 - Platform (MacOS 13.5)
 - Version of EnergyPlus (23.2)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### LINKED ISSUE #10360: Site:GroundReflectance:SnowModifier may use incorrect default for daylighting
Issue overview
--------------
Site:GroundReflectance:SnowModifier has two fields, one for solar gains and one for daylighting, and both default to 1.0.
```
  Site:GroundReflectance:SnowModifier,
    1.0;                     !- Ground Reflected Solar Modifier
    1.0;                     !- Daylighting Ground Reflected Solar Modifier
```

If this object is not present in the input file, then both field default to 1.0 - no problem here.
But, if only the first field is specified, then the second field will be zero (due to incorrect input processing). e.g.
```
  Site:GroundReflectance:SnowModifier,
    1.0;                     !- Ground Reflected Solar Modifier

eio output
! <Site:GroundReflectance:SnowModifier>, Normal, Daylighting {dimensionless}
 Site:GroundReflectance:SnowModifier,   1.000,   0.000
! <Site:GroundReflectance:Snow>,Jan{dimensionless},Feb{dimensionless},Mar{dimensionless},Apr{dimensionless},May{dimensionless},Jun{dimensionless},Jul{dimensionless},Aug{dimensionless},Sep{dimensionless},Oct{dimensionless},Nov{dimensionless},Dec{dimensionless}
 Site:GroundReflectance:Snow,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20
! <Site:GroundReflectance:Snow:Daylighting>,Jan{dimensionless},Feb{dimensionless},Mar{dimensionless},Apr{dimensionless},May{dimensionless},Jun{dimensionless},Jul{dimensionless},Aug{dimensionless},Sep{dimensionless},Oct{dimensionless},Nov{dimensionless},Dec{dimensionless}
 Site:GroundReflectance:Snow:Daylighting,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00
```
This results in reduced daylighting illuminance whenever the snow (on the ground) flag is true. This likely has minimal impact since most winter design days are run without solar, and most epw files have zeroes for snow depth.

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 23.2
 - Revealed by #10295

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file WindowTests.idf and 
[WindowTests-DDaysWithSunAndSnow.idf.txt](https://github.com/NREL/EnergyPlus/files/13844842/WindowTests-DDaysWithSunAndSnow.idf.txt)

 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes #10282. Supersedes #10283?
 - Adds alternative method for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow input for Maximum Outdoor Dry-Bulb Temperature for Supplemental Heater Operation to return default when input field is missing from inputs. Default = 21 C.
 - Fixes incorrect polygon method tag in dxf output. When dxf output is requested without specifying `Report Specifications 1` the dxf file is written using the correct default method `Triangulate3DFace` but the tag in the dxf file was `ThickPolyline`
 - Fixes #10360 .

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by rraustad** [comment]
@mjwitte @yujiex this appears to fill numeric defaults past min-fields. The #10282 defect file reads in 21 C as the default. If this field is entered then that value is used instead of the defaut. I'm not saying all getInputs are set up to read this correctly but in the case of the TUs, the default value is read in correctly.

image

image

If the user enters a value for that field (in this case 17 C) that value will be used instead.

image

**Comment by rraustad** [comment]
For the case of the HeatBalanceAlgorithm, which has no min-fields and this defect file does not have any of the numeric fields, the defaults are filled (just not used since getInput for this object is not set up to read them in unless they are present). 

    HeatBalanceAlgorithm,ConductionTransferFunction;

image

In this case the defaults don't need to be read in since these data are initialized to the defaults in the constructor.

image

    struct HeatBalanceData : BaseGlobalStruct
    {
        Real64 LowHConvLimit = 0.1; // Lowest allowed convection coefficient for detailed model
        Real64 HighHConvLimit = 1000.0;         // upper limit for HConv, mostly used for user input limits in practice. !W/m2-K

    struct HeatBalSurfData : BaseGlobalStruct
    {
        Real64 MaxSurfaceTempLimit = 200.0;            // Highest inside surface temperature allowed in Celsius
        Real64 MaxSurfaceTempLimitBeforeFatal = 500.0; // 2.5 times MaxSurfaceTempLimit

**Comment by rraustad** [comment]
Diff for GroundTempOSCCompactSched:

image

    SurfaceProperty:OtherSideCoefficients,
    N8, \field Period of Sinusoidal Variation
      \note Use with sinusoidal variation to define the time period
      \type real
      \units hr
      \default 24
      \minimum> 0

            state.dataSurface->OSC(OSCNum).Name = state.dataIPShortCut->cAlphaArgs(1);
            state.dataSurface->OSC(OSCNum).SurfFilmCoef = state.dataIPShortCut->rNumericArgs(1);
            state.dataSurface->OSC(OSCNum).ConstTemp = state.dataIPShortCut->rNumericArgs(2); //  This will be replaced if  schedule is used
            state.dataSurface->OSC(OSCNum).ConstTempCoef =
                state.dataIPShortCut->rNumericArgs(3); //  This multiplier is used (even with schedule).  It should normally be 1.0
            state.dataSurface->OSC(OSCNum).ExtDryBulbCoef = state.dataIPShortCut->rNumericArgs(4);
            state.dataSurface->OSC(OSCNum).GroundTempCoef = state.dataIPShortCut->rNumericArgs(5);
            state.dataSurface->OSC(OSCNum).WindSpeedCoef = state.dataIPShortCut->rNumericArgs(6);
            state.dataSurface->OSC(OSCNum).ZoneAirTempCoef = state.dataIPShortCut->rNumericArgs(7);
            state.dataSurface->OSC(OSCNum).SinusoidPeriod = state.dataIPShortCut->rNumericArgs(8);

**Comment by rraustad** [comment]
Diffs for WindowTests:

image

    Site:GroundReflectance:SnowModifier,
    1.0;                     !- Ground Reflected Solar Modifier

There is no object for Site:GroundReflectance:Snow:Daylighting in this example file.

**Comment by nrel-bot-2b** [comment]
@rraustad @myoldmopar it has been 28 days since this pull request was last updated.

File: src/EnergyPlus/InputProcessing/InputProcessor.cc [hunk 1]
```diff
@@ -868,11 +868,7 @@ void InputProcessor::setObjectItemValue(EnergyPlusData &state,
             }
             if (is_AlphaBlank) AlphaBlank()(alpha_index) = true;
         } else if (field_type == "n") {
-            if (within_max_fields) {
-                findDefault(Numbers(numeric_index), schema_field_obj);
-            } else {
-                Numbers(numeric_index) = 0;
-            }
+            findDefault(Numbers(numeric_index), schema_field_obj);
```
Comment (mjwitte) [inline src/EnergyPlus/InputProcessing/InputProcessor.cc:870]: `findDefault` is a bool function. Should check here and set the value to 0 if it returns false.

**Review by mjwitte — COMMENTED** [review]
And we may as well do this for alphas, too?

File: src/EnergyPlus/InputProcessing/InputProcessor.cc (hunk 1 above)
Comment (rraustad) [reply src/EnergyPlus/InputProcessing/InputProcessor.cc:870]: No need to check, Numbers(numeric_index) is set to 0 at top of function and if default does not exist for that field will return false and a 0 for that field.

    bool InputProcessor::findDefault(Real64 &default_value, json const &schema_field_obj)
    {
        auto const find_default = schema_field_obj.find("default");
        default_value = 0;
        if (find_default != schema_field_obj.end()) {
            // get default for autosize, int or real
        }
        return false;
    }

**Comment by rraustad** [comment]
For alpha fields I don't think it's an issue, at least not the same issue we see with Real fields. It's either a required-field or it's not. And if it's not and is a key choice then there should be a default. If there is no default I think that's an issue with the idd not the alpha findDefault function. If it's not a key choice then a non-required field blank should return a blank string (and getInput needs to handle that case?). Worst case here is to add an else at the bottom to "set" default_value to a null string but I don't think that's necessary, because the string passed to this function should already be blank.

    bool InputProcessor::findDefault(std::string &default_value, json const &schema_field_obj)
    {
        auto const find_default = schema_field_obj.find("default");
        if (find_default != schema_field_obj.end()) {
            auto const &default_val = find_default.value();
            if (default_val.is_string()) {
                default_value = default_val.get<std::string>();
            } else {
                if (default_val.is_number_integer()) {
                    i64toa(default_val.get<std::int64_t>(), s);
                } else {
                    dtoa(default_val.get<double>(), s);
                }
                default_value = s;
            }
            if (schema_field_obj.find("retaincase") == schema_field_obj.end()) {
                default_value = Util::makeUPPER(default_value);
            }
            return true;
        }
        return false;
    }

**Comment by mjwitte** [comment]
For alpha fields, it looks like the code won't fill in the default unless it's `within_max_fields `, see line 866
```
    if (field_type == "a") {
            if (!(within_max_fields && findDefault(Alphas(alpha_index), schema_field_obj))) {
                Alphas(alpha_index) = "";
```

**Comment by rraustad** [comment]
I see that. I wonder what to do with lines 876 and 882 to get NumAlphas and NumNumbers right? I think I'll just make the one suggestion now to see what happens.

image

**Comment by rraustad** [comment]
Given the changes to unit tests I'd say this is working as expected.

File: tst/EnergyPlus/unit/InputProcessor.unit.cc [hunk 2]
```diff
@@ -3416,7 +3421,7 @@ TEST_F(InputProcessorFixture, getObjectItem_fan_on_off)
 
     EXPECT_EQ(4, NumAlphas);
     EXPECT_TRUE(compare_containers(
-        std::vector<std::string>({"SUPPLY FAN 1", "FANANDCOILAVAILSCHED", "ZONE EXHAUST NODE", "DX COOLING COIL AIR INLET NODE", "", "", ""}),
+        std::vector<std::string>({"SUPPLY FAN 1", "FANANDCOILAVAILSCHED", "ZONE EXHAUST NODE", "DX COOLING COIL AIR INLET NODE", "", "", "General"}),
```
Comment (rraustad) [inline tst/EnergyPlus/unit/InputProcessor.unit.cc:3423]: Retain case on this last alpha field:

    Fan:OnOff,
      A7 ; \field End-Use Subcategory
           \note Any text may be used here to categorize the end-uses in the ABUPS End Uses by Subcategory table.
           \type alpha
           \retaincase
           \default General

**Comment by mjwitte** [comment]
These eio diffs are significant.
```
- Site:GroundReflectance:SnowModifier,   1.000,   0.000
+ Site:GroundReflectance:SnowModifier,   1.000,   1.000
 ! ,Jan{dimensionless},Feb{dimensionless},Mar{dimensionless},Apr{dimensionless},May{dimensionless},Jun{dimensionless},Jul{dimensionless},Aug{dimensionless},Sep{dimensionless},Oct{dimensionless},Nov{dimensionless},Dec{dimensionless}
  Site:GroundReflectance:Snow,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20
 ! ,Jan{dimensionless},Feb{dimensionless},Mar{dimensionless},Apr{dimensionless},May{dimensionless},Jun{dimensionless},Jul{dimensionless},Aug{dimensionless},Sep{dimensionless},Oct{dimensionless},Nov{dimensionless},Dec{dimensionless}
- Site:GroundReflectance:Snow:Daylighting,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00
+ Site:GroundReflectance:Snow:Daylighting,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20,  0.20
```

There are no results diffs on CI, because the weather file doesn't have any times with snow on the ground and the winter design day has no sun and no snow.

But if you change the winter design day to have snow and solar, then there's a difference in the daylighting illuminance. 
Results for WindowTests-DDaysWithSunAndSnow, develop on the left, this branch on the right.

image  image

**Comment by mjwitte** [comment]
The dxf diffs also reveal a bug:
```
 DXF created from EnergyPlus
 999
 999
-Polygon Action,ThickPolyline
+Polygon Action,TRIANGULATE3DFACE
```
In this case, the drawing is correct, just the info line is wrong, because this line of code is wrong:
https://github.com/NREL/EnergyPlus/blob/6fafbb763fe676012f5db10da7b3e8449f578094/src/EnergyPlus/OutputReports.cc#L532-L533

The drawing action is correctly set to the default Triangulate3DFace, but the print statement is writing the wrong string. With the changes in this branch, `PolygonAction` will never be empty. So, this if can be removed, or at least change the offending line to say "Triangulate3DFace".

**Comment by rraustad** [comment]
Re: DXF diffs: maybe the default used to be ThickPolyline? I did wonder why there were DXF diffs, thanks for checking.

**Comment by rraustad** [comment]
The impact of this change will be on getInput syntax. json has a similar issue getting defaults for missing fields. For example:

    ChillerAbsorption getInput:

        if (NumNums > 16) {
            thisChiller.GeneratorSubcool = state.dataIPShortCut->rNumericArgs(17);
        } else {
            thisChiller.GeneratorSubcool = 1.0;
        }

        if (NumNums > 17) {
            thisChiller.SizFac = state.dataIPShortCut->rNumericArgs(18);
        } else {
            thisChiller.SizFac = 1.0;
        }

should change to:

        thisChiller.GeneratorSubcool = state.dataIPShortCut->rNumericArgs(17);
        thisChiller.SizFac = state.dataIPShortCut->rNumericArgs(18);

because:

    Chiller:Absorption,
      N17, \field Degree of Subcooling in Steam Generator
           \type real
           \units C
           \default 1.0
      N18; \field Sizing Factor
           \type real
           \minimum> 0.0
           \default 1.0

**Comment by mjwitte** [comment]
I wouldn't worry about removing code that sets hard-wired defaults like the absorption chiller. That can go away when `getObjectItem` goes away and it's all replaced with direct json input processing.

**Comment by Myoldmopar** [comment]
@rraustad @mjwitte this is a really exciting change.  I am super happy to see the removal of default values duplicated in both the IDD and the get-input routines.  From the comments above, it seems the EIO diffs are still an issue?  Let me know if I can do anything to help spur this along, as I'd love to get this in.

**Comment by mjwitte** [comment]
> @rraustad @mjwitte this is a really exciting change. I am super happy to see the removal of default values duplicated in both the IDD and the get-input routines. From the comments above, it seems the EIO diffs are still an issue? Let me know if I can do anything to help spur this along, as I'd love to get this in.

I posted a new issue regarding the eio diffs, #10360 . This PR fixes that issue, no further changes required here.

For the dxf diff, it's fixed here and has minimal impact, but it would be good to clean up the offending code per [this comment](https://github.com/NREL/EnergyPlus/pull/10295#issuecomment-1875552978).

Otherwise, this is good to go.

**Comment by rraustad** [comment]
I'm OK with the eio diffs, as expected. And @mjwitte has added to the description as necessary for unanticipated changes revealed by this branch. This looks ready to merge.

**Comment by mjwitte** [comment]
@Myoldmopar Assuming CI comes back same as for https://github.com/NREL/EnergyPlus/commit/d6cfeba3501e35b5cbc6ed587b50bb6d8c1251fd#comments this is ready to merge. Do you want to review anymore?

File: src/EnergyPlus/InputProcessing/InputProcessor.cc (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/InputProcessing/InputProcessor.cc:870]: I'm not always inclined to use the latest fancy C++, but I will just note that C++23 will have std::expected, and it has the potential to be deployed throughout our code some day in the future.  We're years from anything like that, so no reason to discuss it, but feel free to look it up if you are an interested student.

**Review by Myoldmopar — APPROVED** [review]
All good, let's get this in.

**Comment by Myoldmopar** [comment]
One unrelated unit test failure on Mac, otherwise good to go.  Thanks @rraustad and @mjwitte !  Merging this one.

**Comment by Myoldmopar** [comment]
Note (to self mostly) this could cause diffs in some of this morning's other merges.  I'll run regressions locally as needed.

**Pull request merged**